test/e2e/cloudstack_upgrade.go

```diff
@@ -3,160 +3,205 @@
 package e2e
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/aws/eks-anywhere/internal/pkg/api"
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/test/framework"
+	corev1 "k8s.io/api/core/v1"
 )
 
+type cloudStackAPIUpgradeTestStep struct {
+	name         string
+	configFiller api.ClusterConfigFiller
+}
+
 type cloudStackAPIUpgradeTest struct {
 	name string
 	// steps is a list are grouped updates to be applied during a test synchronously.
 	steps []cloudStackAPIUpgradeTestStep
 }
 
-type cloudStackAPIUpgradeTestStep struct {
-	name         string
-	configFiller api.ClusterConfigFiller
-}
-
-func defaultCloudStackAPIUpgradeTestFillers(cloudstack *framework.CloudStack) api.ClusterConfigFiller {
-	return api.JoinClusterConfigFillers(
-		api.ClusterToConfigFiller(
-			api.WithControlPlaneCount(1),
-			api.RemoveAllWorkerNodeGroups(), // This gives us a blank slate
-		),
-		// Add new WorkerNodeGroups
-		cloudstack.WithWorkerNodeGroup("md-0", framework.WithWorkerNodeGroup("md-0", api.WithCount(1))),
-		cloudstack.WithWorkerNodeGroup("md-1", framework.WithWorkerNodeGroup("md-1", api.WithCount(1))),
-		cloudstack.WithRedhat123(),
-	)
+func clusterPrefix(value string, prefix string) string {
+	return fmt.Sprintf("%s-%s", prefix, value)
```

Comment:
When creating multiple clusters in the test, we can't use the same name for the machineConfigs. So, I'm using this to prefix the worker node names with the cluster name.